### PR TITLE
Network upgrade and fork

### PR DESCRIPTION
**Pull request by Andrew-Pohl**
Upgrade to OE 3.2.6, schedules a fork at block 14million.

Updated quickstart to easily update Nodes, Validators, BridgeValidators and bootnodes. Explorer nodes will need a manual DB update since we do not host any snapshots for explorer nodes.

### CODE REVIEW

File: config/spec.json [hunk 1]
```diff
@@ -57,10 +64,11 @@
...
       "0x0000000000000000000000000000000000000004": { "balance": "1", "builtin": { "name": "identity", "pricing": { "linear": { "base": 15, "word": 3 } } } },
-      "0x0000000000000000000000000000000000000005": { "builtin": { "name": "modexp", "activate_at": 0, "pricing": { "modexp": { "divisor": 20 } } } },
-      "0x0000000000000000000000000000000000000006": { "builtin": { "name": "alt_bn128_add", "activate_at": 0, "pricing": { "linear": { "base": 500, "word": 0 } } } },
-      "0x0000000000000000000000000000000000000007": { "builtin": { "name": "alt_bn128_mul", "activate_at": 0, "pricing": { "linear": { "base": 40000, "word": 0 } } } },
-      "0x0000000000000000000000000000000000000008": { "builtin": { "name": "alt_bn128_pairing", "activate_at": 0, "pricing": { "alt_bn128_pairing": { "base": 100000, "pair": 80000 } } } },
+      "0x0000000000000000000000000000000000000005": { "builtin": { "name": "modexp", "pricing": { "0": { "price": { "modexp": { "divisor": 20 } } }, "0xd59f80": { "info": "EIP-2565: ModExp Gas Cost.", "price": { "modexp2565": {} } } } } },
```
Comment (leonprou) [inline config/spec.json:67]: Can you send a reference to the op code changes?

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
-FROM ubuntu:18.04
-
+FROM alpine:3.13.2 AS builder
```
Comment (leonprou) [inline Dockerfile:1]: This is needed just for the build, right? So yeah we don't need PARITY here

File: config/spec.json [hunk 3]
```diff
@@ -39,6 +39,13 @@
...
     "eip1014Transition": "0x38ada7",
     "eip1052Transition": "0x38ada7",
+    "eip1283Transition": "0xd59f80",
+    "eip1344Transition": "0xd59f80",
+    "eip1706Transition": "0xd59f80",
+    "eip1884Transition": "0xd59f80",
+    "eip2028Transition": "0xd59f80",
+    "eip2929Transition": "0xd59f80",
+    "eip2930Transition": "0xd59f80",
     "maxCodeSize": 24576,
```
Comment (leonprou) [inline config/spec.json:49]: This is still relevant for OE?
From what you told me that is configured on the node client now

File: scripts/quickstart.sh [hunk 4]
```diff
@@ -324,6 +327,14 @@ function pullSnapShot {
   echo -e "\nPulling snapshot..."
 
   if [[ $ROLE != explorer ]] ; then
```
Comment (leonprou) [inline scripts/quickstart.sh:329]: So the parity still supported with the new quickstart, right?

File: config/spec.json (hunk 3 above)
Comment (Andrew-Pohl) [reply config/spec.json:49]: I made a mistake here, maxCodeSizeTransition is used to signal the block at which you want to update code size to the value set by maxCodeSize (so will be a fork). Want to bump this up for the fork at 14million or further down the line?

File: Dockerfile (hunk 2 above)
Comment (leonprou) [reply Dockerfile:1]: I suppos we'll get rid of Parity changes after the OE upgrade. IMO need to have a separate backlog for Network

File: scripts/quickstart.sh [hunk 5]
```diff
@@ -40,7 +40,10 @@ REQUIRED_DRIVE_SPACE_MB=15360
 REQUIRED_RAM_MB=1800
 DEFAULT_GAS_ORACLE="https:\/\/ethgasstation.info\/json\/ethgasAPI.json"
 
-SNAPSHOT_NODE="https://node-snapshot.s3.eu-central-1.amazonaws.com/db.tar.gz"
+PARITY_SNAPSHOT="https://node-snapshot.s3.eu-central-1.amazonaws.com/db.tar.gz"
+OE_SNAPSHOT="https://node-snapshot-oe.s3.eu-central-1.amazonaws.com/db.tar.gz"
+
+SNAPSHOT_NODE="$OE_SNAPSHOT"
```
Comment (leonprou) [inline scripts/quickstart.sh:46]: This should de chose by the `CLIEINT`, no?

File: config/spec.json (hunk 1 above)
Comment (leonprou) [reply config/spec.json:67]: @Andrew-Pohl 👼

**Comment by Andrew-Pohl** [comment]
> > > Looks good. I understand now what's going on :) Can you explain the validator upgrade process?
> > 
> > 
> > Sure the upgrade process for node, bootnodes and validators is simple. I have added a new element into the .env file "CLIENT" this is used to tell the quickstart script what build to run OE or parity (defaults to OE). If this key isn't present in an env file (for existing users) it will assume that they are moving from parity->OE. As a result it will delete the existing DB and pull the new OE one from our S3 bucket. The S3 bucket is updated weekly so they will always be within 90k blocks of the head block (quick sync up).
> > For explorer nodes it's a little different since we don't host an archive DB stored on S3. OE have made a tool to upgrade databases which I have tested the quickstart will point users to this.
> > So for non explorer nodes it's simply download the new quickstart and run.
> > For explorer nodes it:
> > 
> > 1. Download new quickstart
> > 2. Stop running docker container
> > 3. Download DB upgrade tool
> > 4. (optional) back up existing DB
> > 5. Run tool on DB
> > 6. run quickstart
> 
> Sounds good, But I think the default for now (without having client) should be parity, so if Validators do not switch to OE without knowing. Need also to explain to them the upgrade process, and that some downtime is anticipated (right?). Idieallly when a validator is switching from parity to OE he would have some `y/N` prompt

Good idea if they don't have CLIENT there I will add a y/n prompt to upgrade.

**Comment by Andrew-Pohl** [comment]
> > > > Looks good. I understand now what's going on :) Can you explain the validator upgrade process?
> > > 
> > > 
> > > Sure the upgrade process for node, bootnodes and validators is simple. I have added a new element into the .env file "CLIENT" this is used to tell the quickstart script what build to run OE or parity (defaults to OE). If this key isn't present in an env file (for existing users) it will assume that they are moving from parity->OE. As a result it will delete the existing DB and pull the new OE one from our S3 bucket. The S3 bucket is updated weekly so they will always be within 90k blocks of the head block (quick sync up).
> > > For explorer nodes it's a little different since we don't host an archive DB stored on S3. OE have made a tool to upgrade databases which I have tested the quickstart will point users to this.
> > > So for non explorer nodes it's simply download the new quickstart and run.
> > > For explorer nodes it:
> > > 
> > > 1. Download new quickstart
> > > 2. Stop running docker container
> > > 3. Download DB upgrade tool
> > > 4. (optional) back up existing DB
> > > 5. Run tool on DB
> > > 6. run quickstart
> > 
> > 
> > Sounds good, But I think the default for now (without having client) should be parity, so if Validators do not switch to OE without knowing. Need also to explain to them the upgrade process, and that some downtime is anticipated (right?). Idieallly when a validator is switching from parity to OE he would have some `y/N` prompt
> 
> Good idea if they don't have CLIENT there I will add a y/n prompt to upgrade.

Added in 98c63d5

File: config/spec.json (hunk 1 above)
Comment (Andrew-Pohl) [reply config/spec.json:67]: @leonprou added in the info 8ccea94

**Pull request merged**